include/boost/url/impl/url_base.ipp

```diff
@@ -270,20 +270,20 @@ set_user(string_view s)
 
 url_base&
 url_base::
-set_user(pct_encoded_view s)
+set_user(decode_view s)
 {
     grammar::detail::copied_strings<
         BOOST_URL_STACK_BYTES> buf(
             this->string());
     s = s.maybe_copy(buf);
```

Comment:
`copied_strings` knows about `decode_view`?

Reply:
oh the other way around, `decode_view` has a `maybe_copy` that knows about `copied_strings`

Reply:
Yes. You implemented it. :)

Reply:
lmao